mediacapture-streams/MediaStream-feature-policy.https.html

```diff
@@ -0,0 +1,97 @@
+<!DOCTYPE html>
+<body>
+    <script src=/resources/testharness.js></script>
+    <script src=/resources/testharnessreport.js></script>
+    <script src=/common/get-host-info.sub.js></script>
+    <script>
+function withIFrame(url, allow)
+{
+    return new Promise(resolve => {
+        let frame = document.createElement('iframe');
+        frame.src = url;
+        frame.allow = allow;
+        frame.onload = () => { resolve(frame); };
+        document.body.appendChild(frame);
+    });
+}
+
+var crossDomain = get_host_info().HTTPS_REMOTE_ORIGIN;
+function runTestInIFrame(allow, crossOrigin)
+{
+    return new Promise(async resolve => {
+        window.addEventListener('message', function handler(event) {
+            resolve(event.data);
+            window.removeEventListener('message', handler);
+            iframe.remove();
+        });
+        let url = location.pathname + "#test";
+        if (crossOrigin)
+            url = crossDomain + url;
+        const iframe = await withIFrame(url, allow);
+    });
+}
+
+async function doTest()
+{
+    let results = { };
+    try {
+        await navigator.mediaDevices.getUserMedia({audio : true});
+        results.audio = true;
+    } catch (e) {
+        results.audio = false;
+    }
+    try {
+        await navigator.mediaDevices.getUserMedia({video : true});
+        results.video = true;
+    } catch (e) {
+        results.video = false;
+    }
+    try {
+        await navigator.mediaDevices.getUserMedia({audio : true, video : true});
+        results.audioVideo = true;
+    } catch (e) {
+        results.audioVideo = false;
+    }
+    return results;
+}
+
+if (location.hash == "#test") {
+    doTest().then(data => {
+        parent.postMessage(data, '*');
+    }, error => {
+        parent.postMessage('error', '*');
+    });
```

Comment:
Since we're already in an `async` function, this could be:
```
try {
  parent.postMessage(await doTest(), '*');
} catch(error) {
  parent.postMessage('error', '*');
}
```